### ISSUE TITLE
Support LoadBearing parameter for IfcStair and IFcStairFlight

### ISSUE THREAD

**Issue by MJB-online** [Autodesk/revit-ifc#275]
Hello all,

Is it possible to support the parameter "LoadBearing" for **IfcStair** and **IfcStairFlight**. We are a producer of prefab concrete stairs, and can imagine situations where stair parts are loaded by other parts (walls, etc.).

Sincerely,
Mark

**Comment by AngelVelezSosa** [comment]
I don't think that parameter is part of stair's definition.  Is there a situation where a stair or stair flight *isn't* load bearing?  Isn't that unexpected?

**Comment by WawanSolihin** [comment]
LoadBearing is defined as a property in IFC4 for Pset_StairCommon only (no LoadBearing for Pset_StairFlightCommon). In this case, you may assign the value and it will get exported for IfcStair if you are using IFC4. In IFC2x3, LoadBearing is not defined in Pset_StairCommon as well as Pset_StairFlightCommon.

**Comment by ArtemHnatko** [comment]
**Properties for IfcStair and IfcStairFlight in IFC2x3 and IFC4
IFC2x3:**
In IFC2x3, the predefined property sets for IfcStair and IfcStairFlight are limited compared to IFC4. The following properties are included:

**IfcStair – Pset_StairCommon**

- Reference
- Status
- IsExternal
- ThermalTransmittance
- FireRating
- AcousticRating

**IfcStairFlight – Pset_StairFlightCommon**

- Reference
- Status
- NumberOfRiser
- NumberOfTreads
- RiserHeight
- TreadLength
- NosingLength
- WalkingLineOffset
- TreadLengthAtOffset
- TreadLengthAtInnerSide
- HasRiser

💡 **Note:** In **IFC2x3**, the **LoadBearing** property is **not defined** for either IfcStair or IfcStairFlight.

**IFC4:**
In IFC4, additional property sets and parameters are available, improving the level of detail in exported data.

**IfcStair – Pset_StairCommon**
(Same as IFC2x3 but with the following additions)

- LoadBearing (New in IFC4, indicates whether the stair is load-bearing)
- AccessibilityPerformance
- FireExit
- HandicapAccessible

**IfcStairFlight – Pset_StairFlightCommon**
(Same properties as in IFC2x3, but LoadBearing is not included)

💡 **Key Difference:** In **IFC4**, the **LoadBearing** property is **available for IfcStair but not for IfcStairFlight**.

![Image](https://github.com/user-attachments/assets/859c4b20-4048-4533-9fe9-8ca3649dc769)
